File: src/main/java/org/acme/util/Tokens.java
/*
 * Created on 14 Oct 2014
 */

package org.acme.util;

public class Tokens {

    private String theValues;
    private char theChar;

    public Tokens(
            final String aString,
            final char aChar,
            final int aOffset)
            throws Exception {
        //msg("aString =" + aString);
        //msg("aOffset =" + aOffset);
        if (aOffset > aString.length()) {
            throw new Exception("Tokens: offset length error");
        }
        theChar = aChar;
        theValues = aString.substring(aOffset, aString.length());
    }

    public String next() {
        if (theValues == null) {
            return null;
        }
        final int myIndexOf = theValues.indexOf(theChar);
        //msg("myIndexOf =" +myIndexOf);
        if (myIndexOf == -1) {
            final String myRet = theValues;
            theValues = null;
            return myRet;
        }
        final String myRet = theValues.substring(0, myIndexOf);
        theValues = theValues.substring(myIndexOf + 1, theValues.length());
        return myRet;

    }

    private static void msg(final String s) {
        System.out.println(Tokens.class.getName() + "> " + s);
    }
}

File: src/main/resources/META-INF/resources/js/_obj/tokens.js




 function Tokens(aString, aChar, aOffset) {
    if ((typeof aOffset) !== 'number') {
        throw 'aOffset not a number ' + aOffset;
    }
    //
    if (aChar.length !== 1) {
        throw 'cahar length error: > ' + aChar.length;
    }
    if (aOffset > aString.length) {
        throw 'offset length error: ' + aOffset + ' > ' + aString.length;
    }
    //
    this.__theChar = aChar;
    this.__theValues = aString.substring(aOffset, aString.length);
};




Tokens.prototype.next = function () {
    //alert('next');
    if (this.__theValues === null) {
        return null;
    }

    var myIndexOf = this.__theValues.indexOf(this.__theChar);
    if (myIndexOf === -1) {
        var myRet = this.__theValues;
        this.__theValues = null;
        return myRet;
    }
    var myRet = this.__theValues.substring(0, myIndexOf);
    this.__theValues = this.__theValues.substring(myIndexOf + 1, this.__theValues.length);
    return myRet;
};


